node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/identity/AthenzCredentialsMaintainer.java

```diff
@@ -0,0 +1,281 @@
+import com.yahoo.vespa.hosted.node.admin.component.Environment;
+import com.yahoo.vespa.hosted.node.admin.configserver.noderepository.NodeSpec;
+import com.yahoo.vespa.hosted.node.admin.util.PrefixLogger;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Set;
+
+import static java.util.Collections.singleton;
+
+/**
+ * A maintainer that is responsible for providing and refreshing Athenz credentials for a container.
+ *
+ * @author bjorncs
+ */
+public class AthenzCredentialsMaintainer {
+
+    private static final Duration EXPIRY_MARGIN = Duration.ofDays(1);
+    private static final Duration REFRESH_PERIOD = Duration.ofDays(1);
+    private static final Path CONTAINER_SIA_DIRECTORY = Paths.get("/var/lib/sia");
+
+    private static final ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+
+    private final boolean enabled;
+    private final PrefixLogger log;
+    private final String hostname;
+    private final Path trustStorePath;
+    private final Path privateKeyFile;
+    private final Path certificateFile;
+    private final AthenzService containerIdentity;
+    private final URI ztsEndpoint;
+    private final Clock clock;
+    private final ServiceIdentityProvider hostIdentityProvider;
+    private final IdentityDocumentClient identityDocumentClient;
+    private final InstanceCsrGenerator csrGenerator;
+    private final AthenzService configserverIdentity;
+    private final String zoneRegion;
+    private final String zoneEnvironment;
+
+    public AthenzCredentialsMaintainer(String hostname,
+                                       Environment environment,
+                                       ServiceIdentityProvider hostIdentityProvider) {
+        ContainerName containerName = ContainerName.fromHostname(hostname);
+        Path containerSiaDirectory = environment.pathInNodeAdminFromPathInNode(containerName, CONTAINER_SIA_DIRECTORY);
+        this.enabled = environment.isNodeAgentCertEnabled();
+        this.log = PrefixLogger.getNodeAgentLogger(AthenzCredentialsMaintainer.class, containerName);
+        this.hostname = hostname;
+        this.containerIdentity = environment.getNodeAthenzIdentity();
+        this.ztsEndpoint = environment.getZtsUri();
+        this.configserverIdentity = environment.getConfigserverAthenzIdentity();
+        this.csrGenerator = new InstanceCsrGenerator(environment.getCertificateDnsSuffix());
+        this.trustStorePath = environment.getTrustStorePath();
+        this.privateKeyFile = getPrivateKeyFile(containerSiaDirectory, containerIdentity);
```

Comment:
Some things to consider:
* The path must be mounted into the container for it to be "visible" to `node-admin`. `/var/lib/sia` is not mounted for tenants: https://github.com/vespa-engine/vespa/blob/4a055d7cd4c56c868b9fd03a96abc495f76ab68e/node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java#L321:L361
* `node-admin` runs as `yahoo` (`host-admin` runs a `root`). So these directories need to be made writable by `yahoo`.